src/Layer.js

```diff
@@ -399,7 +399,9 @@ export default class Layer {
     _paintLayer () {
         if (this._viz && this._viz.colorShader) {
             this._renderLayer.setViz(this._viz);
-            this.renderer.renderLayer(this._renderLayer);
+            this.renderer.renderLayer(this._renderLayer, {
+                zoomLevel: this.map.getZoom()
+            });
             if (this.isAnimated() || this._fireUpdateOnNextRender || !util.isSetsEqual(this._oldDataframes, new Set(this._renderLayer.getActiveDataframes()))) {
```

Comment:
I agree with the second part `const framesHaveChanged = !util.isSetsEqual(this._oldDataframes, this._getActiveDataFrames()))`. But not with the first one (`_getActiveDataFrames() {...`), I don't agree with that because the only real thing that the method is doing (creating a set) is being hidden from the caller and it is not an implementation detail, I also don't think that extracting that behavior (of 1 line) is worth of 2 extra lines

Reply:
👍 I just saw it was being used twice, in the if and to get the old frames, but fair enough!